storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -2106,19 +2111,27 @@ static bool innodb_init_param(void) {
 
   srv_buf_pool_chunk_unit = 134217728;
   srv_buf_pool_instances = 1;
-  estimate_memory = xtrabackup_prepare && redo_memory != 0 &&
-                    redo_frames != 0 && !xtrabackup_use_memory_set &&
+  if (xtrabackup_incremental_dir) {
+    real_redo_memory = incremental_redo_memory;
+    real_redo_frames = incremental_redo_frames;
+  } else {
+    real_redo_memory = redo_memory;
+    real_redo_frames = redo_frames;
+  }
+
+  estimate_memory = xtrabackup_prepare && real_redo_memory != 0 &&
+                    real_redo_frames != 0 && !xtrabackup_use_memory_set &&
                     xtrabackup_use_free_memory_pct != 0;
 
   if (estimate_memory) {
     ulint free_memory_total = xtrabackup::utils::host_free_memory();
     ulint free_memory_usable =
         (free_memory_total * xtrabackup_use_free_memory_pct) / 100;
-    ulint mem =
-        buf_pool_size_align(redo_memory + (redo_frames * UNIV_PAGE_SIZE));
+    ulint mem = buf_pool_size_align(real_redo_memory +
+                                    (real_redo_frames * UNIV_PAGE_SIZE));
     if (mem > (ulint)xtrabackup_use_memory) {
-      xb::info() << "Got estimation from backup: " << redo_memory
-                 << " bytes for parsing and " << redo_frames
+      xb::info() << "Got estimation from backup: " << real_redo_memory
+                 << " bytes for parsing and " << real_redo_frames
```

Comment:
little below we have this:
```
if ((long)redo_frames > innobase_open_files) {
      xb::info() << "Setting innobase_open_files to: " << redo_frames;
      innobase_open_files = (long)redo_frames;
}
```
shoudln't these be changed to real_redo_frames? (I think it is better to unset redo_frames/redo_memory after it is read to real_redo_frames/memory, may be use some max VALUES and add some assertions.  Idea is to prevent unintended usage)

Reply:
and add ut_ad(real_redo/frames != UINT64_MAX)

Reply:
another questions I have is why are we using redo_frames as files :thinking:

Reply:
> and add ut_ad(real_redo/frames != UINT64_MAX)

This is to make sure we are not exceeding innodb_buffer_pool_size max value?
If so, what about `ut_ad(real_redo_memory + (real_redo_frames * UNIV_PAGE_SIZE) < UINT64_MAX)`

Reply:
Ack. Will add